### PR TITLE
TargetState

### PR DESCRIPTION
**Pull request by alicekuznetsov**
@TaylerUva @Alenguye582 

bros...
While working on this, I thought that a HashMap with the desired `TargetState` would be the cleanest way of storing the preset values, but I wasn't sure how we'd want to store those groups of measures in it.
I could think of 3 implementations I enjoyed but I'm not sure which one is the best 😀. Ultimately, it doesn't matter, as I'll just write a shooter method that accepts whatever we pick [such as `setShooterToPreset(Matrix<N3, N1>)`] but I think we'll run into this issue later if we continue using Measures

1. Measure array size isn't specified 
`public static Map<TargetState, Measure[]> TARGET_TO_PRESETS; ` 

2. 2 sets of pairs (tuples aren't in our IDE and I couldn't find an alternative, let me know if you do bc i would love that)
`public static Map<TargetState, Pair<Measure<Angle>, Pair<Measure<Velocity<Angle>>, Measure<Velocity<Angle>>>>> TARGET_TO_PRESETS;`

3. Convert the Measures to doubles and store them in a Matrix
`public static Map<TargetState, Matrix<N3, N1>> TARGET_TO_PRESETS;`

(I also considered making a separate object like `public ThreeSpecificMeasures(Measure<Angle>[] shooterAngle, Measure<Velocity<Angle>> leftVelocity, Measure<Velocity<Angle>> rightVelocity)` and using that as the key value but I figured that would be excessive)

### CODE REVIEW

**Comment by TaylerUva** [comment]
I like 'public static Map<TargetState, Measure[]> TARGET_TO_PRESETS;' however then it is on the user to make sure the order is correct. Object idea prevents that specific issue at the cost of some increased complexity.

I'm leaning toward Object as my #1 and Measure array as #2

We could also make 3 maps: 1 for angle one for left and one for right which may be the simplest implementation?

**Comment by alicekuznetsov** [comment]
I don't like the idea of 3 separate maps just for the sake of keeping them together. 
I'll go with the object implementation :)

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -276,6 +290,64 @@ public static class constShooter {
...
+        constShooter.LEFT_SHUFFLE_VELOCITY, constShooter.RIGHT_SHUFFLE_VELOCITY);
+    public static final ShooterPositionGroup PREP_SUB = new ShooterPositionGroup(constShooter.PIVOT_SUB_ANGLE,
+        constShooter.LEFT_SUB_VELOCITY, constShooter.RIGHT_SUB_VELOCITY);
+    public static final ShooterPositionGroup PREP_SPIKE = new ShooterPositionGroup(constShooter.PIVOT_SPIKE_ANGLE,
+        constShooter.LEFT_SPIKE_VELOCITY, constShooter.RIGHT_SPIKE_VELOCITY);
+    public static final ShooterPositionGroup PREP_WING = new ShooterPositionGroup(constShooter.PIVOT_WING_ANGLE,
+        constShooter.LEFT_WING_VELOCITY, constShooter.RIGHT_WING_VELOCITY);
+
+  }
+
```
Comment (TaylerUva) [inline src/main/java/frc/robot/Constants.java]: Is there a need to create constants only to pass them to another constant? Why not just store only in this object?

**Review by TaylerUva — DISMISSED** [review]

**Comment by alicekuznetsov** [comment]
Closes #59

**Review by AngelaKimmy — APPROVED** [review]

**Pull request merged**